optimum/neuron/generation/utils.py

```diff
@@ -82,6 +78,91 @@ class NeuronGenerationMixin(GenerationMixin):
     learn more about decoding strategies refer to the [text generation strategies guide](../generation_strategies).
     """
 
+    @staticmethod
+    def _initialize_attention(
+        model_kwargs,
+        num_padding_values,
+        batch_size,
+        device,
+        is_encoder_decoder,
+    ):
+        """Initializes the appropriate attention mask -- encoder-decoder models use `decoder_attention_mask`"""
+        if is_encoder_decoder:
+            # One 1 for decoder_start_token_id, 0s for the currently-unfilled locations in the past_key_values tensor,
+            # 1s for the actual input_ids
+            decoder_attention_mask = torch.cat(
+                [
+                    torch.zeros((batch_size, num_padding_values), dtype=torch.int32),
+                    torch.ones((batch_size, 2), dtype=torch.int32),
```

Comment:
Shouldn't the ones come first in the concatenation?

Reply:
Intuitively, it should be the opposite. I do not use `_initialize_attention` in my case as the pkv is initialized inside the compiled "encoder". I only move `_initialize_attention` out of `_update_model_kwargs_for_xla_generation` to be able to reuse `_update_attention` and make it easier to override `_update_model_kwargs_for_xla_generation`.